src/main/java/uk/gov/justice/hmpps/prison/service/AppointmentsService.java

```diff
@@ -93,11 +116,38 @@ public List<CreatedAppointmentDetails> createAppointments(@NotNull @Valid final
         final var appointmentsWithRepeats = withRepeats(appointments.getRepeat(), flattenedDetails);
         assertThatAppointmentsFallWithin(appointmentsWithRepeats, appointmentTimeLimit());
 
+        EventStatus eventStatus = eventStatusRepository.findById(EventStatus.SCHEDULED_APPROVED)
+            .orElseThrow(() -> new RuntimeException("Event status not recognised."));
+        final AgencyLocation prison = agencyLocationRepository.findById(agencyId)
+            .orElseThrow(() -> new RuntimeException("Prison not found"));
+        final AgencyInternalLocation location = agencyInternalLocationRepository.findById(defaults.getLocationId())
+            .orElseThrow(() -> new RuntimeException("Location not found"));
+
         final var createdAppointments = appointmentsWithRepeats.stream().map(a -> {
-                final var appointmentId = bookingRepository.createAppointment(a, defaults, agencyId);
+                final var booking = offenderBookingRepository.findById(a.getBookingId())
+                    .orElseThrow(() ->
+                        new HttpClientErrorException(HttpStatus.BAD_REQUEST, "A BookingId does not exist in your caseload")
+                    );
+                if (!agencyId.equals(booking.getLocation().getId())) {
+                    throw new HttpClientErrorException(HttpStatus.BAD_REQUEST, "A BookingId does not exist in your caseload");
```

Comment:
Keeping existing error behaviour

Reply:
`HttpClientErrorException` is a Spring exception thrown by WebClient - seems a bit confusing to be throwing it for different reasons.

Reply:
Yep , i could try replacing it with a BadRequest exception next time